component/metrics/scrape/types.go

```diff
@@ -0,0 +1,286 @@
+package scrape
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/go-kit/log/level"
+	"github.com/hashicorp/hcl/v2"
+	common_config "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/rfratto/gohcl"
+)
+
+var emptyAuthorization = common_config.Authorization{}
+var emptyBasicAuth = common_config.BasicAuth{}
+var emptyOAuth2 = &common_config.OAuth2{}
+
+const bearer string = "Bearer"
+
+// Config holds all of the attributes that can be used to configure a scrape
+// component.
+type Config struct {
```

Comment:
Oh my, this object is big. I'd recommend trimming it down for the MVP or splitting into multiple children blocks so we don't have to prefix fields like `tls_config_insecure_skip_verify`.

Reply:
I trimmed it down to multiple pointer-type optional blocks on https://github.com/grafana/agent/pull/1850/commits/6709f3436b05f19d5667dfd3e962fd4edf80a2d6. It looks cleaner, simplifies checking for invalid configs and doesn't need prefixes, so I'd consider it an improvement.